pkg/rbd/controllerserver.go

```diff
@@ -91,7 +91,16 @@ func (cs *ControllerServer) validateVolumeReq(req *csi.CreateVolumeRequest) erro
 func parseVolCreateRequest(req *csi.CreateVolumeRequest) (*rbdVolume, error) {
 	// TODO (sbezverk) Last check for not exceeding total storage capacity
 
-	rbdVol, err := getRBDVolumeOptions(req.GetParameters())
+	// MultiNodeWriters are accepted but they're only for special cases, and we skip the watcher checks for them which isn't the greatest
+	// let's make sure we ONLY skip that if the user is requesting a MULTI Node accessbile mode
+	ignoreMultiWriterEnabled := true
+	for _, am := range req.VolumeCapabilities {
+		if am.GetAccessMode().GetMode() != csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER {
```

Comment:
what about explicitly checking `VolumeCapability_AccessMode_MULTI_NODE_SINGLE_WRITER` and `VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER`, because there are two single node capabilities in csi spec:
```go
const (
        VolumeCapability_AccessMode_UNKNOWN VolumeCapability_AccessMode_Mode = 0
        // Can only be published once as read/write on a single node, at
        // any given time.
        VolumeCapability_AccessMode_SINGLE_NODE_WRITER VolumeCapability_AccessMode_Mode = 1
        // Can only be published once as readonly on a single node, at
        // any given time.
        VolumeCapability_AccessMode_SINGLE_NODE_READER_ONLY VolumeCapability_AccessMode_Mode = 2
        // Can be published as readonly at multiple nodes simultaneously.
        VolumeCapability_AccessMode_MULTI_NODE_READER_ONLY VolumeCapability_AccessMode_Mode = 3
        // Can be published at multiple nodes simultaneously. Only one of
        // the node can be used as read/write. The rest will be readonly.
        VolumeCapability_AccessMode_MULTI_NODE_SINGLE_WRITER VolumeCapability_AccessMode_Mode = 4
        // Can be published as read/write at multiple nodes
        // simultaneously.
        VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER VolumeCapability_AccessMode_Mode = 5
)
```

Reply:
ack, I had that planned for a follow up:  https://github.com/ceph/ceph-csi/pull/239/commits/6c473d26d6e6ec973b1a7d855c2aaa88fed7c059#diff-170d986ec944158cc00a94f7e60f989eR106

Let me take a look at handling that case now instead if you're good with it.